script/constructron.lua

```diff
@@ -707,13 +745,60 @@ me.conditions = {
                 trunk[item.name] = (trunk[item.name] or 0) + item.count
             end
         end
+
+        unfufilled_requests = {}
+        unfufilled_requests_count = 1
+
+
         for i = 1, constructron.request_slot_count do ---@cast i uint
             local request = constructron.get_vehicle_logistic_slot(i)
             if request then
                 if not (((trunk[request.name] or 0) >= request.min) and ((trunk[request.name] or 0) <= request.max)) then
-                    return false -- condition is not met
+                    -- Create table of unfufilled items
+                    table.insert(unfufilled_requests, unfufilled_requests_count, request.name)
+                    unfufilled_requests_count = unfufilled_requests_count + 1
+                end
+            end
+        end
+
+        log("Optimistic Items:")
+        if #global.optimistic_logistics_items > 0 then
+
+            for i, k in pairs(global.optimistic_logistics_items) do
+                log("   " .. k)
+            end
+        end
+        log("End optimistic items")

```

Comment:
This seems only to be for debugging. Would probably be good to wrap this in a debug enabled check to only interate this when actually debugging.

Or remove this from here altogether and rather do it after the global variable has been filled.